iot-e2e-tests/common/pom.xml

```diff
@@ -100,6 +96,20 @@
                     <compilerArgument>-XDignore.symbol.file</compilerArgument>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <version>2.2</version>
+                <executions>
+                    <execution>
+                        <goals>
+                            <!--This plugin will generate a jar file that contains all the test code so that the android
+                            test code can reference it.-->
+                            <goal>test-jar</goal>
```

Comment:
previously, this test code was in a normal jar, not a test jar. This extra step is now necessary to package this new test project